lib/galaxy/datatypes/proteomics.py

```diff
@@ -92,6 +233,277 @@ def display_peek(self, dataset):
         return self.make_html_table(dataset, column_names=self.column_names)
 
 
+@build_sniff_from_prefix
+class Dta(TabularData):
+    """dta
+    The first line contains the singly protonated peptide mass (MH+) and the
+    peptide charge state separated by a space. Subsequent lines contain space
+    separated pairs of fragment ion m/z and intensity values.
+    """
+    file_ext = "dta"
+    comment_lines = 0
+
+    def set_meta(self, dataset, **kwd):
+        column_types = []
+        data_row = []
+        data_lines = 0
+        if dataset.has_data():
+            with open(dataset.file_name, 'r') as dtafile:
+                for line in dtafile:
+                    data_lines += 1
+
+        # Guess column types
+        for cell in data_row:
+            column_types.append(self.guess_type(cell))
+
+        # Set metadata
+        dataset.metadata.data_lines = data_lines
+        dataset.metadata.comment_lines = 0
+        dataset.metadata.column_types = ['float', 'float']
+        dataset.metadata.columns = 2
+        dataset.metadata.column_names = ['m/z', 'intensity']
+        dataset.metadata.delimiter = " "
+
+    def sniff_prefix(self, file_prefix):
+        has_data = False
+        for line in file_prefix.line_iterator():
+            line = line.strip().split(" ")
+            if len(line) != 2:
+                return False
+            try:
+                line = [float(_) for _ in line]
+            except ValueError:
+                return False
+            if not all(_ >= 0 for _ in line):
+                return False
+            has_data = True
+        return has_data
+
+
+@build_sniff_from_prefix
+class Dta2d(TabularData):
+    """
+    dta2d: files with three tab/space-separated columns.
+    The default format is: retention time (seconds) , m/z , intensity.
+    If the first line starts with '#', a different order is defined by the the
+    order of the keywords 'MIN' (retention time in minutes) or 'SEC' (retention
+    time in seconds), 'MZ', and 'INT'.
+    Example: '#MZ MIN INT'
+    The peaks of one retention time have to be in subsequent lines.
+
+    Note: sniffer detects (tab or space separated) dta2d files with correct
+    header, wo header seems to generic
+    """
+    file_ext = "dta2d"
+    comment_lines = 0
+
+    def _parse_header(self, line):
+        if len(line) != 3 or len(line[0]) < 3 or not line[0].startswith("#"):
+            return None
```

Comment:
Why not return False here?

Reply:
Could change this, but I found it more appealing to return a list (if successful) or None (otherwise). I also check for `None` in the remaining code.